SYCL/Basic/reqd_work_group_size.cpp

```diff
@@ -12,6 +12,167 @@
+  Q.submit([&](handler &CGH) {
+    CGH.parallel_for<ReqdWGSizeVariadic<Kind, ReqdWGSizeX>>(
+        GlobalRange,
+        [=](item<1>) [[sycl::reqd_work_group_size(ReqdWGSizeX)]] {});
+  });
+}
+
+template <int... ReqdWGSizes>
+int runPositiveTest(queue &Q, range<sizeof...(ReqdWGSizes)> GlobalRange,
+                    range<sizeof...(ReqdWGSizes)> LocalRange) {
+  std::string TestCaseName =
+      "Test case ReqdWGSizeVariadic " +
+      GetRangeStr(range<sizeof...(ReqdWGSizes)>{ReqdWGSizes...}) + " & " +
+      GetRangeStr(LocalRange) + " (Positive with local range)";
+  try {
+    submitTest<TestKind::PositiveWithLocal, ReqdWGSizes...>(Q, GlobalRange,
+                                                            LocalRange);
+    Q.wait_and_throw();
+    return 0;
+  } catch (nd_range_error &E) {
+    std::cerr << TestCaseName
+              << " failed: unexpected nd_range_error exception: " << E.what()
+              << std::endl;
+    return 1;
+  } catch (runtime_error &E) {
+    std::cerr << TestCaseName
+              << " failed: unexpected runtime_error exception: " << E.what()
+              << std::endl;
+    return 1;
+  } catch (...) {
+    std::cerr << TestCaseName << " failed: something unexpected has been caught"
+              << std::endl;
+    return 1;
+  }
+}
+
+template <int... ReqdWGSizes>
+int runPositiveTest(queue &Q, range<sizeof...(ReqdWGSizes)> GlobalRange) {
+  std::string TestCaseName =
+      "Test case ReqdWGSizeVariadic " +
+      GetRangeStr(range<sizeof...(ReqdWGSizes)>{ReqdWGSizes...}) +
+      " (Positive without local range)";
+  try {
+    submitTest<TestKind::PositiveWithoutLocal, ReqdWGSizes...>(Q, GlobalRange);
+    Q.wait_and_throw();
+    return 0;
+  } catch (nd_range_error &E) {
+    std::cerr << TestCaseName
+              << " failed: unexpected nd_range_error exception: " << E.what()
+              << std::endl;
+    return 1;
+  } catch (runtime_error &E) {
+    std::cerr << TestCaseName
+              << " failed: unexpected runtime_error exception: " << E.what()
+              << std::endl;
+    return 1;
+  } catch (...) {
+    std::cerr << TestCaseName << " failed: something unexpected has been caught"
+              << std::endl;
+    return 1;
+  }
+}
+
+template <int... ReqdWGSizes>
+int runNegativeTest(queue &Q, range<sizeof...(ReqdWGSizes)> GlobalRange,
+                    range<sizeof...(ReqdWGSizes)> LocalRange) {
+  std::string TestCaseName =
+      "Test case ReqdWGSizeVariadic " +
+      GetRangeStr(range<sizeof...(ReqdWGSizes)>{ReqdWGSizes...}) + " & " +
+      GetRangeStr(LocalRange) + " (Negative)";
+  try {
+    submitTest<TestKind::Negative, ReqdWGSizes...>(Q, GlobalRange, LocalRange);
+    Q.wait_and_throw();
+    std::cerr << TestCaseName << " failed: no exception has been thrown\n";
+    return 1; // We shouldn't be here, exception is expected
+  } catch (nd_range_error &E) {
+    return 0;
+  } catch (runtime_error &E) {
```

Comment:
'runtime_error' is deprecated.

Reply:
You're right! All uses of it has been removed.